meta-mender-tegra/scripts/mender_tegra_test.py

```diff
@@ -0,0 +1,263 @@
+        rootfs_part = None
+        if boot_slot == '0':
+            rootfs_part = 'A'
+        elif boot_slot == '1':
+            rootfs_part = 'B'
+
+        if rootfs_part != None:
+            result = conn.run(f'grep -h {rootfs_part} /etc/mender/mender.conf /var/lib/mender/mender.conf | cut -d: -f2 | cut -d, -f1 | tr -d \'" \'')
+            result = conn.run(f'df -h | grep {result.stdout.strip()}')
+            if result.return_code != 0:
+                raise RuntimeError("Boot and Rootfs Partition Mismatch detected")
+        else:
+            raise RuntimeError("Cannot Identify Rootfs Partition Slot")
+
+    def verify_slot_change_across_reboots(self):
+        prev_boot_slot = self.nvbootctrl_current_slot()
+        self.reboot()
+        boot_slot = self.nvbootctrl_current_slot()
+        if boot_slot != prev_boot_slot:
+           raise RuntimeError("Boot slot changed from {} to {} across reboots".format(prev_boot_slot,boot_slot))
+
+    def verify_slot_change_across_mender_update(self):
+        prev_boot_slot = self.nvbootctrl_current_slot()
+        self.reboot()
+        boot_slot = self.nvbootctrl_current_slot()
+        if boot_slot == prev_boot_slot:
+           raise RuntimeError("Boot slot not changed from {} to {} after mender update reboot".format(prev_boot_slot,boot_slot))
+
+    def do_single_mender_update(self, rollback_flag):
+        # Connecting to the device
+        self.wait_for_device()
+        # mender install 
+        self.mender_install()
+        # Make sure the sentinel file is not present before reboot for successful boot cases.
+        # sentinel file required for rollback cases
+        if rollback_flag:
+            self.add_sentinel_file()
+        else:
+            self.remove_sentinel_file()
+
+        # Check Partition Mismatch
+        self.check_partition_mismatch()
+        # Check slot changed after a mender update followed by reboot.
+        self.verify_slot_change_across_mender_update()
+
+        # rollback_flag is set true to replicate rollback case, false to replicate successful mender update case.
+        # Mender commit is redundant in case of cboot(no harm in commiting or not), but we need to make sure we don't mender commit when testing rollback case for uboot.
+        # Since that would lead to partition mismatch.
+        # In conclusion for uboot, mender commit is required for a successful mender update and not required to test the rollback case.
+        if not rollback_flag:
+            self.mender_commit()
+
+    def check_rollback(self):
+        boot_slot = self.nvbootctrl_current_slot()
+
+        # nvidia gives 7 retries before switching boot partition
+        for i in range(7):
+            print(f'Starting reboot {i} to test rollback case')
+            # Check Partition Mismatch
+            self.check_partition_mismatch()
+            # check the current boot_slot
+            if boot_slot != self.nvbootctrl_current_slot():
+                self.count += 1 
```

Comment:
Should we exit the loop here?  Do we need `count`?  If so we should give it a more descriptive name.